pkg/operator/controller/ingress_status.go

```diff
@@ -0,0 +1,102 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// syncIngressControllerStatus computes the current status of ic and therefrom
+// updates the syncIngressControllerStatus resource.
+func (r *reconciler) syncIngressControllerStatus(deployment *appsv1.Deployment, ic *operatorv1.IngressController) error {
+	selector, err := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil {
+		return fmt.Errorf("ingress controller deployment %s/%s has invalid spec.selector: %v", deployment.Namespace, deployment.Name, err)
```

Comment:
Random thought for later: we own that field, so if it's invalid, this might be a pretty severe problem unless it's an outside mutation we correct later during a resync

Reply:
Good point. Thanks for the insight. When you say "correct later during a resync", it's my understanding that `ingresscontroller.status.selector` would be updated to reflect the mutated  `deployment.Spec.Selector` as apposed to the Operator ensuring `deployment.Spec.Selector` reflects the value of `ingresscontroller.status.selector`. Is my understanding correct?

Reply:
Sounds right — however, I believe such a change to the deployment would cause the operator to detect a diff and overwrite the mutated deployment selector field, which would then propagate back to the ingresscontroller status. Could do a quick test to confirm

Reply:
>  I believe such a change to the deployment would cause the operator to detect a diff and overwrite the mutated deployment selector field

@ironcladlou that's not what I observe. If I mutate the `Deployment` selector, a reconciliation is triggered and the mutated selector is observed in the associated `IngressController`.

Reply:
`deploymentConfigChanged` does not compare the deployments' pod selectors, so the operator will not revert modifications to the same, and `syncClusterIngressStatus` will reflect whatever pod selector is in the deployment to the ingresscontroller's status.

Reply:
@ironcladlou @Miciah you can review the details of my deployment selector testing [here](https://gist.githubusercontent.com/danehans/09315e07537eca731cd79bf44d3535b2/raw/2e6131d4ac784902f8c365e2dfc15b0a9785456d/gistfile1.txt) that confirm my understanding.

Reply:
I expect the deployment controller deleted the old pods (with the `ingress.operator.openshift.io/ingress-controller-deployment=default` label) and create new pods (with the `foo=bar` label) when you modified the deployment, right? And the next time the operator updates the deployment, it will revert the changes, so the pods will all get recreated again.  In the meantime, the anti-affinity policy will not work properly, but I think everything else will be fine.

So if I understand correctly, if an administrator changes the labels, anti-affinity breaks and pods get recreated, but nothing will go horribly broken—which is pretty good considering that outside mutations are not something we recommend or have made efforts to accommodate!

I notice you changed both the deployment's label selector as well as the pod template's labels to match.  If you hadn't done that, the breakage would be more severe.  On the other hand, if the administrator modifies the deployment that _we_ manage, _and_ modifies it to an invalid state, the administrator just asking for problems.